pkg/notifiers/slack/handle_request.go

```diff
@@ -189,7 +220,20 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
 	return nil
 }
 
-func (n *Notifier) UpdateSlackMessage(ctx context.Context, slackClient *slack.Client, log *zap.SugaredLogger, rev access.Reviewer, req access.Request, rule rule.AccessRule, dbRequestor *identity.User) error {
+type UpdateSlackMessageOpts struct {
+	Ctx           context.Context
```

Comment:
In Go, it is not idiomatic to put context inside a struct, instead, it should always be passed as the first argument in a function when needed.

I would also preference only including "data" in the opts for something like this, and keeping slackSlient and log as parameters in the function call
```
UpdateSlackMessage(ctx context.Context, slackClient *slack.Client, log *zap.SugaredLogger, opts UpdateSlackMessageOpts)
```

Reply:
Some content from [ardanlabs]( https://www.ardanlabs.com/blog/2019/09/context-package-semantics-in-go.html#:~:text=It's%20an%20idiom%20in%20Go,pointer%20semantics%20should%20be%20used.&text=Every%20function%20that%20accepts%20a,copy%20of%20the%20interface%20value.)